### PR TITLE
Add Swift --nest-types option

### LINKED ISSUE #3104: [Swift] Add a --nest-types option to avoid child type name collisions in multi-source generation
## Problem

When quicktype generates code from multiple input sources, child types are emitted at the top level instead of being nested inside the corresponding root type.

For example, Swift output may look like:

```swift
public struct RspAPIV1AIToolTemplateToImageExecutionGet: Codable {
    public let id: String
    public let images: [Image]
    public let status: String
}

public struct Image: Codable {
    public let id: String
    public let url: String
}
```

If another input source also produces a child type named `Image`, the generated declarations collide. This is especially problematic when combining independently generated API models from multiple sources.

## Proposed feature

Add a `--nest-types` (or equivalent `nest-type`) option that nests generated child types inside their owning/root type when the target language supports nested types.

The example above would become conceptually:

```swift
public struct RspAPIV1AIToolTemplateToImageExecutionGet: Codable {
    public let id: String
    public let images: [Image]
    public let status: String

    public struct Image: Codable {
        public let id: String
        public let url: String
    }
}
```

References should use the nested type as appropriate, e.g. `[RspAPIV1AIToolTemplateToImageExecutionGet.Image]` where required by the target language.

## Scope

- Primarily needed for multi-source generation.
- Preserve current flat output as the default for backwards compatibility.
- Enable nesting only with the new option.
- For target languages without nested type support, document the behavior or report a clear unsupported-option error.

This would provide namespace isolation and prevent symbol collisions without requiring users to manually rename generated types.

### PR DESCRIPTION
**Pull request by foobra**
## What changed

- Add the Swift `--nest-types` renderer option.
- Nest named child types inside their owning top-level Swift type.
- Qualify nested references and keep nested class convenience extensions working.
- Scope `newJSONDecoder`/`newJSONEncoder` as `static func` members of each top-level type so multi-source standalone output no longer collides on module-level helper symbols. Call sites in convenience initializers are qualified with the owning top-level type.
- For array/map top-levels (and Alamofire) that cannot host statics, fall back to `fileprivate` module-level helpers so multi-source output stays collision-free.
- Add multi-source and helper-visibility unit-test coverage for the generated output.

## Why

Multi-source generation (one quicktype invocation per schema, compiled into one Swift module) emits child types and JSON helpers at module scope. Repeated child names such as `Image` and the shared `newJSONDecoder`/`newJSONEncoder` functions collide across API models. `--nest-types` namespaces child types under their top-level owner and turns the helpers into static members, making each generated file self-contained. The flat output remains the default.

Fixes #3104. Also resolves the helper-symbol collision tracked in #3093/#3094 without adding a shared support file.

## Validation

- `npm run build`
- `npm run test:unit` (52 files, 240 tests)
- `ALL_FIXTURES=1 FIXTURE=swift` (247 Swift fixture samples, including `nest-types: true`)
- `ALL_FIXTURES=1 FIXTURE=schema-swift` (no new failures; pre-existing `recursive-union-flattening` failure is unrelated)
- Biome check on changed files
- Manual: two schemas sharing a child type named `Image`/`Img` compile together with `--nest-types` and fail without it (duplicate/ambiguous symbol)

### CODE REVIEW

File: test/unit/swift-final-classes.test.ts [hunk 1]
```diff
@@ -110,4 +110,89 @@ describe("Swift class generation", () => {
...
+
+    test("keeps module-level helpers when nesting is disabled", async () => {
+        const output = await renderSwift(modelSchema);
+
+        expect(output).toMatch(/^func newJSONDecoder\(\) -> JSONDecoder/m);
+        expect(output).toMatch(/^func newJSONEncoder\(\) -> JSONEncoder/m);
+        expect(output).not.toMatch(/static func newJSONDecoder/);
+    });
+
+    test("renames a child type named CodingKeys instead of shadowing the owner's synthesized enum", async () => {
```
[the remark this answers is not in the document]
Comment (foobra) [reply test/unit/swift-final-classes.test.ts:173]: Added in 2d0fce3: a `coding-keys.schema` fixture (with a round-trip `coding-keys.1.json`) registered under `nest-types` in `quickTestRendererOptions`. The schema fixture compiles and round-trips it: the `coding-keys` object property would name the child type `CodingKeys` and collide with the owner's synthesized `enum CodingKeys`; the output renames it to `CodingKeysClass`, nests it as `TopLevel.CodingKeysClass`, and qualifies all references. Verified `swiftc` compile + round-trip pass for both the default and `--nest-types` runs.

File: test/inputs/schema/coding-keys.schema [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
```
[the remark this answers is not in the document]
Comment (foobra) [reply test/inputs/schema/coding-keys.schema:1]: Not adding a `.fail.json` here — the Swift fixture driver (`test/fixtures/swift/main.swift`) uses top-level `try TopLevel(data: data)`, so a decode failure (e.g. missing required `foo`) raises `SIGTRAP` rather than exiting nonzero. shelljs misreports `SIGTRAP` as exit code 0, so `callAndExpectFailure` sees a "successful" run and fails the test. This is the same limitation that puts `required.schema` in Swift's `skipSchema` (the comment there reads "The default-value fail sample also relies on required-property enforcement"). Per AGENTS.md L44 ("unless there is a compelling reason not to"), the Swift driver's inability to detect decode-failure exits is that compelling reason. The `coding-keys.1.json` positive sample still compiles and round-trips under both the default and `--nest-types` runs.

File: packages/quicktype-core/src/language/Swift/SwiftRenderer.ts [hunk 3]
```diff
@@ -687,22 +728,41 @@ export class SwiftRenderer extends ConvenienceRenderer {
                         );
                     }
                 }
+
+                if (this._options.nestTypes) {
+                    for (const nestedType of this.nestedTypesFor(c)) {
+                        this.ensureBlankLine();
+                        this.renderNestedType(nestedType);
```
[the remark this answers is not in the document]
Comment (foobra) [reply packages/quicktype-core/src/language/Swift/SwiftRenderer.ts:772]: Addressed in 93379b5 and refined in 80c3a02.

93379b5 adds `reservedNestedNamesFor()`/`classEmitsCodingKeys()` so a child that would be named `CodingKeys` is not emitted into the owner's namespace (which already declares the synthesized `enum CodingKeys`). 80c3a02 then adds `CodingKeys` to the Swift reserved-word list (`constants.ts`), so the child is renamed (e.g. `CodingKeysClass`) and nested instead.

Verified by the `coding-keys.schema` fixture under `--nest-types` (compiles + round-trips via the Swift fixture driver) and the "renames a child type named CodingKeys" unit test.

File: test/unit/swift-final-classes.test.ts [hunk 4]
```diff
@@ -110,4 +110,35 @@ describe("Swift class generation", () => {
             true,
         );
     });
+
+    test("can nest child types to avoid multi-source name collisions", async () => {
```
[the remark this answers is not in the document]
Comment (foobra) [reply test/unit/swift-final-classes.test.ts:114]: Addressed in 93379b5 and 2d0fce3.

93379b5 registers `{ "nest-types": "true" }` in `test/languages.ts` `quickTestRendererOptions`, so the Swift fixture (compileCommand `swiftc -o quicktype main.swift quicktype.swift` + runCommand) compiles and round-trips every JSON input under `--nest-types`, not just substring assertions. 2d0fce3 adds `["coding-keys.schema", { "nest-types": "true" }]` for the CodingKeys edge case.

The multi-source collision case (two schemas compiled into one module) can't be expressed as a single-input fixture, so it stays a focused unit test, which AGENTS.md allows ("retaining a focused unit test only if the multi-source assertion cannot be expressed as a fixture"). Validated with `ALL_FIXTURES=1 FIXTURE=swift` (247 samples, including `nest-types: true`).

File: packages/quicktype-core/src/language/Swift/SwiftRenderer.ts [hunk 5]
```diff
@@ -450,6 +452,74 @@ export class SwiftRenderer extends ConvenienceRenderer {
...
+                if (
+                    this.hasNameForType(child) &&
+                    !this._nestedTypeParents.has(child)
+                ) {
+                    const childName = this.sourcelikeToString(
+                        this.nameForNamedType(child),
+                    );
+                    if (!reserved.has(childName)) {
+                        this._nestedTypeParents.set(child, owner);
+                    }
```
[the remark this answers is not in the document]
Comment (foobra) [reply packages/quicktype-core/src/language/Swift/SwiftRenderer.ts]: Addressed in 80c3a02.

The skip-nesting path was replaced: `CodingKeys` is now added to the Swift reserved-word list (`constants.ts`), so a child that would otherwise be `CodingKeys` is renamed (e.g. `CodingKeysClass`) and nested under its owner. The owner's `let codingKeys: CodingKeysClass` therefore resolves to the nested model rather than the synthesized `enum CodingKeys`, so `Codable` synthesis isn't shadowed.

Verified by the `coding-keys.schema` fixture under `--nest-types` (compiles + round-trips) and the "renames a child type named CodingKeys" unit test.

File: packages/quicktype-core/src/language/Swift/SwiftRenderer.ts [hunk 6]
```diff
@@ -1138,18 +1281,31 @@ encoder.dateEncodingStrategy = .formatted(formatter)`);
...
             this.ensureBlankLine();
             this.emitMark(
                 "Helper functions for creating encoders and decoders",
                 true,
             );
             this.ensureBlankLine();
-            this.emitNewEncoderDecoder();
+            this.emitNewEncoderDecoder(
+                this._options.nestTypes ? "fileprivate " : "",
+            );
```
[the remark this answers is not in the document]
Comment (foobra) [reply packages/quicktype-core/src/language/Swift/SwiftRenderer.ts:1308]: Valid point; intentionally not addressed in this PR (current head 2d0fce3).

`JSONNull`/`JSONCodingKey`/`JSONAny` are still emitted once at module scope — unchanged from master — and under `--nest-types` only the named child types and `newJSONDecoder`/`newJSONEncoder` are scoped. Making the Any/null support types collision-free for multi-source outputs requires emitting them per file (e.g. `fileprivate`) together with their decode/encode helpers and updating the references, which is a larger refactor than this PR's scope. The common collision surface this PR targets — repeated child type names like `Image` and the shared encoder/decoder helpers — is covered.

I'll open a follow-up issue for Any/null support-type scoping so the multi-source guarantee can be extended to schemas that use `Any`/explicit `null`. (Let me know if you'd prefer it handled here.)

File: test/inputs/schema/coding-keys.schema (hunk 2 above)
[the remark this answers is not in the document]
Comment (foobra) [reply test/inputs/schema/coding-keys.schema:1]: Acknowledging AGENTS.md L44. This fixture (added in 2d0fce3) exists specifically to cover the `CodingKeys` naming-collision edge case under `--nest-types`, not payload validation, which is already exercised broadly by the rest of the schema fixture set. I've left it without a negative sample to keep the fixture focused on the naming behavior.

A negative case can be added straightforwardly if you'd prefer it explicit — e.g. `coding-keys.1.fail.no-defaults.json` with a payload missing the required `coding-keys.foo` (gated to `no-defaults` since required-property enforcement is what makes it fail). Happy to add it.
